crates/task-impls/src/transactions.rs

```diff
@@ -232,114 +232,102 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, V: Versions> TransactionTask
-                self.builder_timeout,
-                self.auction_results_provider
-                    .fetch_auction_result(block_view),
-            )
-            .await
-            {
-                let auction_result = maybe_auction_result
-                    .map_err(|e| warn!("Failed to get auction results: {e:#}"))
-                    .unwrap_or_default(); // We continue here, as we still have fallback builder URL
-
-                let mut futures = Vec::new();
+    ) -> Result<PackedBundle<TYPES>> {
+        ensure!(
+            !self
+                .upgrade_lock
+                .decided_upgrade_certificate
+                .read()
+                .await
+                .as_ref()
+                .is_some_and(|cert| cert.upgrading_in(block_view)),
+            "Not requesting block because we are upgrading",
+        );
 
-                let mut builder_urls = auction_result.clone().urls();
-                builder_urls.push(self.fallback_builder_url.clone());
+        let start = Instant::now();
 
-                for url in builder_urls {
-                    futures.push(async_timeout(
-                        self.builder_timeout.saturating_sub(start.elapsed()),
-                        async {
-                            let client = BuilderClientMarketplace::new(url);
-                            client.bundle(*block_view).await
-                        },
-                    ));
-                }
+        let maybe_auction_result = async_timeout(
+            self.builder_timeout,
+            self.auction_results_provider
+                .fetch_auction_result(block_view),
+        )
+        .await
+        .context("Timeout while getting auction result")?;
+
+        let auction_result = maybe_auction_result
+            .map_err(|e| warn!("Failed to get auction results: {e:#}"))
+            .unwrap_or_default(); // We continue here, as we still have fallback builder URL
+
+        let mut futures = Vec::new();
+
+        let mut builder_urls = auction_result.clone().urls();
+        builder_urls.push(self.fallback_builder_url.clone());
+
+        for url in builder_urls {
+            futures.push(async_timeout(
+                self.builder_timeout.saturating_sub(start.elapsed()),
+                async {
+                    let client = BuilderClientMarketplace::new(url);
+                    client.bundle(*parent_view, parent_hash, *block_view).await
+                },
+            ));
+        }
 
-                let mut bundles = Vec::new();
+        let mut bundles = Vec::new();
 
-                for bundle in join_all(futures).await {
-                    match bundle {
-                        Ok(Ok(b)) => bundles.push(b),
-                        _ => continue,
-                    }
-                }
+        for bundle in join_all(futures).await {
+            match bundle {
+                Ok(Ok(b)) => bundles.push(b),
+                _ => continue,
```

Comment:
Out of scope for this, so feel free to ignore, but it'd be nice to have some logging of the timeout or error here.